papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,498 @@
+Operations can have preconditions and postconditions that define their eligibility, e.g. the existence of an input file in a job's workspace or a key in the job document (as shown in the above snippet) can be a precondition that must be met before an operation can be executed, or a postcondition that indicates an operation is complete. However, this type of conditional workflow can be inefficient when sequential workflows are coupled with an HPC scheduler interface, because the user must log on to the HPC and submit the next operation after the previous operation is complete. This encourages large operations which are not modular and do not accurately represent the individual units of the work-flow limiting signac-flow's utility.
+
+The concept of a group, implemented by the ``FlowGroup`` class and ``FlowProject.make_group`` interface, allows users to combine multiple operations into a group. Submitting a group allows signac-flow to dynamically resolve preconditions and postconditions of operations as each operation is executed, making it possible to combine separate operations (e.g. for simulation and analysis and plotting) into a single submission script with the expectation that all will execute despite later operations depending on the former. Furthermore, groups are aware of directives and can properly combine the directives of their constituent operations to specify resources and quantities like walltime whether executing in parallel or serial.
+
+.. code-block:: python
+
+    from flow import FlowProject
+
+    new_group = FlowProject.make_group(
+        name="new_group")
+
+    @new_group.with_directives(
+        {"ngpu": 2.0,
+         "walltime": lambda j: j.sp.size * 4})
+    @FlowProject.post.true("foo")
+    @FlowProject.operation
+    def foo(job):
+        job.doc.foo = True
+
+    @new_group
+    @FlowProject.pre.true("foo")
+    @FlowProject.post.true("bar")
+    @FlowProject.operation
+    def bar(job):
+        job.doc.bar = True
+
+Groups also allow for specifying multiple machine specific resources (CPU v GPU) with the same operation. An operation can have unique directives for each group it is in. By associating an operation's directives with respect to a specific group, groups can represent distinct compute environments such as a local workstation or a remote supercomputing cluster.
+
+.. code-block:: python
+
+    from flow import FlowProject
+
+    cpu_env = FlowProject.make_group(name="cpu")
+    gpu_env = FlowProject.make_group(name="gpu")
+
+
+    @cpu_env.with_directives({"np": 48})
+    @gpu_env.with_directives({"ngpu": 4})
+    @FlowProject.operation
+    def expensive_operation(job):
+        # expensive computation for either
+        # CPU or GPU here
+        pass
+
+Users also frequently work with multiple jobs at once in a consistent manner.
```

Comment:
what do you mean by "consistent manner"?

Reply:
I had meant that the tasks were amenable to a structured workflow.